### PR TITLE
updated config file name for Python 3.6

### PR DESCRIPTION
**Pull request by lee218llnl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/python/package.py [hunk 1]
```diff
@@ -148,6 +148,15 @@ def install(self, spec, prefix):
         make()
         make('install')
 
+        self.sysconfigfilename = '_sysconfigdata.py'
+        if spec.satisfies('@3.6:'):
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/python/package.py]: Can you add a commented URL pointing the reader to a description of the issue being addressed here?

Comment (lee218llnl) [inline var/spack/repos/builtin/packages/python/package.py]: I don't know if there is a good URL describing the change. I just noticed the file name changed when some of our users reported a build problem with our 3.6.0 python.

Comment (davydden) [inline var/spack/repos/builtin/packages/python/package.py]: maybe just add a quick in-code comment describing what's happening.

Comment (lee218llnl) [inline var/spack/repos/builtin/packages/python/package.py]: I put a comment in there indicating the that config file name changed as of 3.6.0. I'm not sure if you wanted something more detailed, but there's not a whole lot more I can add anyway.

**Pull request merged**